### PR TITLE
mono: 4.0.1 -> 4.0.2.5 + add ability to build mono from git using

### PR DESCRIPTION
**Pull request by obadz**
...the stable version for bootstrapping.

(I needed Mono from Git to get this fix https://github.com/mono/mono/pull/1654 for this bug  https://github.com/fsharp/fsharp/issues/426)

### CODE REVIEW

**Comment by obadz** [comment]
cc @lethalman

**Comment by obadz** [comment]
Throwing in a couple of F# packages

**Comment by obadz** [comment]
bump

File: pkgs/development/compilers/mono/default.nix [hunk 1]
```diff
@@ -1,73 +1,203 @@
-{ stdenv, fetchurl, bison, pkgconfig, glib, gettext, perl, libgdiplus, libX11, callPackage, ncurses, zlib, withLLVM ? false, cacert }:
+{ stdenv
```
Comment (lethalman) [inline pkgs/development/compilers/mono/default.nix]: Please don't put a single package per line. Put 5-6 at least, it's annoying to have such a long preamble.

**Comment by lethalman** [comment]
You really need this mono git to be in nixpkgs rather than your own local machine?

**Comment by obadz** [comment]
I don't ever *need* something to be in nixpkgs but it seems common practice to include various version of the same package in nixpkgs when that's useful.  Seems like it could be useful to other to be able to build both stable and unstable versions of Mono.  That's especially true now given that stable version contains a nasty bug: https://github.com/fsharp/fsharp/issues/426#issuecomment-114983418

**Pull request closed without merging**